content-services/content-sources-frontend
# Fixes 2533: Add red hat repositories UI

## Summary

- Adds Red hat repositories UI to the content table

Additional Information: 
- This needs to be merged alongside it's sister backend PR found [here](https://github.com/content-services/content-sources-backend/pull/435).

## Testing steps

Locally
- After checking out both this branch, and the backend branch [here](https://github.com/content-services/content-sources-backend/pull/435), make sure to run:  make repos-import  &&
go run cmd/external-repos/main.go nightly-jobs; this will populate the Red Hat repositories table. 
- After not to long the repositories should be introspected and you should be able to open the package modal, and close it, confirming that you stay on the correct "Red Hat" table, and the url params stay on "?origin=red_hat
- If you leave the api to run long enough (takes around 20 minutes from my experience) you'll see the Red Hat repositories snapshot, open the snapshot modal and confirm the same functionality as above. 
- With your tab open on the repositories page, and Red Hat repositories selected under the filter options, click on the popular repositories tab, and then back, the table should remain on the red_hat repositories page. << This functionality and the saving of it through reset can be discussed further, I simply left it this way to start the discussion. 
- When switching between Custom and Red Hat repositories, there is also code that can be uncommented (or removed) to reset the filters upon context change if so desired. << Feed back is also needed on desired outcome here.
- While on the RedHat repositories, refresh the page, the page should return the correct toggled button and remain.
